### PR TITLE
Docs for timeline UI (win)

### LINKED ISSUE #4666: Write documentation on Timeline UI feature
### 💻 Environment

Platform: Windows

### 📒 Description
Describe how Timeline UI works and place it in docs/windows.

### PR DESCRIPTION
**Pull request by atsvirchkova**
### 🕶️ Types of changes

Updating windows docs.

### 🤯 List of changes
main-view.md
timeline-view.md

### 👫 Relationships

 Closes #4666 

### 🔎 Review hints
- Do to docs/windows folder.
- Open main-windows.md
- There should be Timeline view part and a link to timeline-view.md document.
- Click the link and see the document.

### CODE REVIEW

File: docs/win/timeline-view.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+# Timeline view
+
+Timeline view is a control which displays time entries in graphical manner as intervals whose placement and length correponds to time entry start and end times. It also includes tracked activity visualization.
```
Comment (skel35) [inline docs/win/timeline-view.md:4]: typo: corre*s*ponds

**Review by skel35 — CHANGES_REQUESTED** [review]
Thanks for the doc!
My main point of feedback is that we could write a bit more - Timeline is probably the most complex part of the app UI and the doc should make it easier to understand for the outsider.
Things I think are worth mentioning:
- TimelineViewModel is written in the ReactiveUI style, and link the docs about ReactiveUI viewmodels
- It supports user interaction such as zooming, resizing TEs, creating new TEs in multiple ways
- You mentioned `toggl_on_timeline` but it's not clear enough that the visualization always happens as a reaction to this library event. `ConvertTimeEntriesToBlocks` is a key method where the magic happens and is worth mentioning too.
- Please also describe, at least briefly, how time entry blocks are laid out onto the main view - an ItemsControl with a Canvas as a panel template, and a location of elements defined through viewmodels. 

Minor notes:
- update the table of contents in `/docs/win/index.md`
- "Turning Record activity is a setting" - needs rephrasing. Did you mean `"Record activity" is a setting` ?

**Comment by atsvirchkova** [comment]
Thank you for feedback!  I'll expand the document to explain to mention the listed points.

File: docs/win/timeline-view.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Timeline view is a control which displays time entries in graphical manner as intervals whose placement/length corresponds to time entry start/end time. It also includes tracked activity visualization.
+
+The implementation can be found in `Timeline.xaml`and `Timeline.xaml.cs` in `ui/views` and `TimelineViewModel.cs` in `ui/ViewModels`. We stick to the [RX](http://introtorx.com/Content/v1.0.10621.0/01_WhyRx.html) concept in this whole feature implementation and use [ReactiveUI](https://www.reactiveui.net/docs/handbook/view-models/) library for view models.
+
+"Record activity" is a setting which allows Toggl to track active windows on user's PC and then show this information in the Activity column.
+
+Main view is a grid, which has 3 columns:
+1) Time - displays the time in hours. 
+2) Time entries - displays time entry block whose placement and height corresponds to the time entry start and end time.
+3) Activity - displays tracked activity as 15 minutes chunks. It's possible to see the list of activities hanging the mouse over the activity block.
```
Comment (skel35) [inline docs/win/timeline-view.md:13]: hanging -> hovering

**Comment by atsvirchkova** [comment]
Updated the doc once more, now it contains some more info on Timeline view. Please, look once more and say what else is worth mentioning.

File: docs/win/timeline-view.md (hunk 2 above)
Comment (skel35) [reply docs/win/timeline-view.md:13]: 15 minutes -> 15-minute

File: docs/win/timeline-view.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+Timeline view is a control which displays time entries in graphical manner as intervals whose placement/length corresponds to time entry start/end time. It also includes tracked activity visualization.
+
+The implementation can be found in `Timeline.xaml`and `Timeline.xaml.cs` in `ui/views` and `TimelineViewModel.cs` in `ui/ViewModels`. We stick to the [RX](http://introtorx.com/Content/v1.0.10621.0/01_WhyRx.html) concept in this whole feature implementation and use [ReactiveUI](https://www.reactiveui.net/docs/handbook/view-models/) library for view models.
+
+"Record activity" is a setting which allows Toggl to track active windows on user's PC and then show this information in the Activity column.
+
+Main view is a grid, which has 3 columns:
+1) Time - displays the time in hours. 
+2) Time entries - displays time entry block whose placement and height corresponds to the time entry start and end time.
```
Comment (skel35) [inline docs/win/timeline-view.md:12]: block -> blocks

File: docs/win/timeline-view.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+### Changing start and end time
+
+User can change start or end time of the time entry using two ways: using the edit view or through resizing. There are `Thumb` controls on the top and bottom of each time entry block, which can be grabbed and dragged vertically. There is an exception for very small blocks (less than `TimelineConstans.MinResizableTimeEntryBlockHeight` in `Resources folder`) - they cannot be resized but only changed though the edit view.
+
+The implementation can be found in `TimelineTimeEntryBlock.xaml`and `TimelineTimeEntryBlock.xaml.cs` in `ui/controls`.
+
+### Create new time entry
+
+A new time entry can be created using click and drag functionality. Start time of the new time entry will match to the left mouse button down point, and end time - to the mouse up point. If user makes just a click or very little drag (less than 2 px) a time entry of one hour long will be created.
```
Comment (skel35) [inline docs/win/timeline-view.md:39]: >Start time of the new time entry will match to the left mouse button down point, and end time - to the mouse up point

not necessarily, can be vice versa 😄

File: docs/win/timeline-view.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+The implementation can be found in `Timeline.xaml`and `Timeline.xaml.cs` in `ui/views`.
+
+## Time entry blocks placing algorithm
+
+All the logic for calculating time entry blocks location on `Canvas` consists in `TimelineViewModel.ConvertTimeEntriesToBlocks`. The main complication is about resolving overlapping time entries, so that minimum number of columns is used. To solve this problem a greedy algorithm described [here](https://www.geeksforgeeks.org/minimum-halls-required-for-class-scheduling/) is used.
```
Comment (skel35) [inline docs/win/timeline-view.md:45]: consists in -> is contained in

**Review by skel35 — CHANGES_REQUESTED** [review]
Good job! Looks much better now.
I think all the main things are covered now, and there's no need to go into many details.
You could also mention the gaps, but that's optional as their implementation is in principle similar to time entry blocks implementation.

I've left a few comments on the biggest choice of words or grammar mistakes, otherwise looks good.

**Comment by atsvirchkova** [comment]
Thanks, fixed once more.

**Review by skel35 — APPROVED** [review]
👍

**Pull request merged**